Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -145,6 +149,30 @@ - (void)paste:(id)sender
     [super paste:sender];
 }
 
+- (void)setFrame:(CGRect)frame
+{
+    BOOL isNonEmpty = (self.width > 0.f && self.height > 0.f);
+    BOOL didChangeSize = !CGSizeEqualToSize(frame.size, self.frame.size);
+
+    [super setFrame:frame];
+
+    if (didChangeSize && isNonEmpty) {
+        [self.textViewPasteDelegate textViewDidChangeSize];
+    }
+}
+
+- (void)setBounds:(CGRect)bounds
+{
+    BOOL isNonEmpty = (self.width > 0.f && self.height > 0.f);
+    BOOL didChangeSize = !CGSizeEqualToSize(bounds.size, self.bounds.size);
+
+    [super setBounds:bounds];
+
+    if (didChangeSize && isNonEmpty) {
+        [self.textViewPasteDelegate textViewDidChangeSize];
```

Comment:
Whenever the text view changes (e.g. because your input wraps to a second line) we want to adjust the scroll position.